### ISSUE TITLE
[P2][Sourcing][Accessibility][Pretest] [Multi combobox] If a value is selected from the list (arrow navigation through the list and exiting with TAB), the value is automatically deleted without error message. Value entry is only possible via the list and selection of a list entry with ENTER (list is closed) or SPACE (list remains open).

### ISSUE THREAD

**Issue by I543348** [SAP/fundamental-ngx#10823]
#### Is this a bug, enhancement, or feature request?
Bug

#### Describe your proposal.
If a value is selected from the list (arrow navigation through the list and exiting with TAB), the value is automatically deleted without error message. Value entry is only possible via the list and selection of a list entry with ENTER (list is closed) or SPACE (list remains open).

Expectation: value should be retained if matching and correct 

#### Can you handle that on the application side
No

#### Which versions of Angular and Fundamental Library for Angular are affected? Please, specify the exact version. (If this is a feature request, use current version.)

Angular: 15
core platform: 0.43.14

**Comment by nikolay-kolarov** [comment]
Hi @I543348,

I do not get it. When an item from the list is selected, then it gets tokenized and the tokens are the list of selected values in the MultiInput, so the value is not lost but moved to a token.
If the value is not matching to a list item, then it is up to the application to decide if it will transform to a token or not.
So could you please provide more info and maybe example what exactly you expect to happen (e.g. when I type XXX in this sample, I want ....)? 

@droshev , @InnaAtanasova  - FYI

Thanks,
Nikolay

**Comment by KaptaanBarbosa** [comment]
> Hi @I543348,
> 
> I do not get it. When an item from the list is selected, then it gets tokenized and the tokens are the list of selected values in the MultiInput, so the value is not lost but moved to a token. If the value is not matching to a list item, then it is up to the application to decide if it will transform to a token or not. So could you please provide more info and maybe example what exactly you expect to happen (e.g. when I type XXX in this sample, I want ....)?
> 
> @droshev , @InnaAtanasova - FYI
> 
> Thanks, Nikolay

there could be another case where in case value is not there in existing list we might want to call remote api and update the list.

**Please reopen this ticket** and check we don't need to delete/replace entered text by user if value is not there in existing list we can get it from DB also via API call.

**Comment by nikolay-kolarov** [comment]
Hello @KaptaanBarbosa ,

The ticket is opened, we just need more info in order to process it.
Could you please provide the exact user interaction where the text does not have to be deleted, so that this ticket can be processed?
For example: I type XXX, then this is not available in the list, then I call and API in the background ... And then this happens, but we expect something else.

At the moment I found one issue when testing on https://ng-15-downport--fundamental-ngx.netlify.app/#/platform/multi-input : when I  start typing something with a letter that is on the list e.g. "Toronto" then it is not possible, but if I want to type "Berlin", then I can. Is this the problem that you are referring to or it is something else?


Thanks,
Nikolay

**Comment by sagar7259** [comment]
Hi @nikolay-kolarov 

Lets take the same example https://ng-15-downport--fundamental-ngx.netlify.app/#/platform/multi-combobox,

*  Here if user want to search "Apple" the search results will be displaying only apple and not pineapple.
Pineapple also contains text "apple" which should also be displayed.
*  If user want to enter random text like "qwe", "zqw" etc., the text is not at all allowed to be typed and error message "Invalid entry" is shown.

Basically user should be allowed to enter free text and should not restrict to enter only the characters available in the list in correct order. Because user will not be knowing all the list items present within the UI.

Secondly by this you can also avoid the error message displaying/reading for the users simply by allowing the user to enter anything they want and showing the search results as "Zero".

Regards,
Sagar

**Comment by nikolay-kolarov** [comment]
Hello @sagar7259 ,

Thanks for the details.
With regards to your first point about apple and pineapple this is according to UX and the default filtering recommended is "starts with per term", which means that the suggestions are filtered if there is a word that starts with the entered value. So if the suggestion was "Pine Apple" it should be matched for "apple". What you are looking for is "contains" filter and I do not know if there is a mechanism for changing the default filtering. And if it doesn't then this is an enhancement request and not a bug.
As a reference, you can check the samples in UI5 https://ui5.sap.com/#/entity/sap.m.MultiInput:
![image](https://github.com/SAP/fundamental-ngx/assets/17043004/0c9b52dc-f9db-4cd7-a993-39c6571ca9f8)
So if this is requested by the current scenario (@I543348), please arrange a feature request with @droshev.

About the second part, this is a UX decision to optimize the entering of the data in the MultiComboBox. The control is designed to accept only items that are included in the list and that is why you cannot enter another value and the user input of wrong data is prevented. If there is a need to have another values then the MultiInput is more suitable as control to be used.
Please cross check UI5 as a reference where it is the same: https://ui5.sap.com/#/entity/sap.m.MultiComboBox/sample/sap.m.sample.MultiComboBox

What I gave as statement for the two points is also clarified in the Fiori Guidelines: https://experience.sap.com/fiori-design-web/multi-combobox/

![image](https://github.com/SAP/fundamental-ngx/assets/17043004/98c6731b-695a-44ce-8ed9-8e41cd0caf36)
![image](https://github.com/SAP/fundamental-ngx/assets/17043004/88a9aa24-dfe3-4678-b130-2ee9d4a834df)

And with this I think the issue can be closed.

PS: For my previous comment I was looking in the MultiInput and that is why I pointed out some questions there. But as we are talking for MultiComboBox then this is not relevant.

Best Regards,
Nikolay

**Comment by nikolay-kolarov** [comment]
@droshev  - please review my comment and close the item if you agree.

**Comment by I543348** [comment]
we are good here  we will also document it for Sourcing side.
Thanks.
